### PR TITLE
Full height chatbot (and other visual improvements)

### PR DESCRIPTION
**Pull request by aliabid94**
ChatInterfaces (and other Blocks optionally) take full screen height. Rearranged interface a little:

Recording 2024-02-01 at 16 48 17

Closes: https://github.com/gradio-app/gradio/issues/4001

### CODE REVIEW

File: gradio/blocks.py [hunk 1]
```diff
@@ -504,6 +504,7 @@ def __init__(
         css: str | None = None,
         js: str | None = None,
         head: str | None = None,
+        fill_vertical_space: bool = False,
```
Comment (abidlabs) [inline gradio/blocks.py:507]: What about just renaming this parameter `full_height`. Its less verbose and clearer than `fill_vertical_space` imo

File: gradio/chat_interface.py [hunk 2]
```diff
@@ -112,6 +112,7 @@ def __init__(
             theme=theme,
             js=js,
             head=head,
+            fill_vertical_space=True,
```
Comment (abidlabs) [inline gradio/chat_interface.py:115]: This should definitely be a parameter in `ChatInterface` to allow users to opt-out

File: gradio/chat_interface.py [hunk 3]
```diff
@@ -73,9 +73,9 @@ def __init__(
         analytics_enabled: bool | None = None,
         submit_btn: str | None | Button = "Submit",
         stop_btn: str | None | Button = "Stop",
-        retry_btn: str | None | Button = "🔄  Retry",
-        undo_btn: str | None | Button = "↩️ Undo",
-        clear_btn: str | None | Button = "🗑️  Clear",
+        retry_btn: str | None | Button = "Retry",
+        undo_btn: str | None | Button = "Undo",
+        clear_btn: str | None | Button = "Clear",
```
Comment (abidlabs) [inline gradio/chat_interface.py:78]: I'm not a huge fan of removing the emojis. For two reasons: now the buttons look too monotonous to me, and we lose the benefits of the buttons being immediately clear and good for i18n. Why remove them?

**Comment by abidlabs** [comment]
Very cool @aliabid94! This was a requested feature a while back so I added it to the issue. I do have some concerns about the changes to `gr.ChatInterface`, see above

File: gradio/chat_interface.py (hunk 3 above)
Comment (pngwn) [reply gradio/chat_interface.py:78]: I don't love the emojis, but I would like us to keep some 'symbols' for the reasons you mentioned. I think some nice clean icons would be better.

**Comment by pngwn** [comment]
This seems to contain some of @freddyaboulton's changes from a previous PR. I think if you rebase/ merge main it should clean itself up.

File: gradio/chat_interface.py (hunk 3 above)
Comment (abidlabs) [reply gradio/chat_interface.py:78]: True now that we support icons in buttons, we could have some cleaner icons instead by default

File: gradio/blocks.py (hunk 1 above)
Comment (aliabid94) [reply gradio/blocks.py:507]: full_height is not accurate though. All apps take full height with the footer. This parameter specifically means that top level components, by default, expand to fill the vertical space. Open to other names, cc: @pngwn

File: gradio/chat_interface.py (hunk 3 above)
Comment (aliabid94) [reply gradio/chat_interface.py:78]: ok will revert but I think the emojis are ugly

**Comment by aliabid94** [comment]
> This seems to contain some of @freddyaboulton's changes from a previous PR. I think if you rebase/ merge main it should clean itself up. 

I'm bad at git. Gonna close this PR and try again

**Pull request closed without merging**